R/integrations.R

```diff
@@ -178,18 +214,37 @@ basicSeuratProc <- function(count.matrix, vars.to.regress=NULL, verbose=TRUE, do
 #'
 #' @export
 #'
-velocityInfoConos <- function(cms.list, con, n.odgenes = 2e3, verbose = TRUE, min.max.cluster.average.emat=0.2, min.max.cluster.average.nmat=0.05, min.max.cluster.average.smat=0.01) {
+velocityInfoConos <- function(cms.list, con, clustering=NULL, groups=NULL, n.odgenes=2e3, verbose=TRUE, min.max.cluster.average.emat=0.2, min.max.cluster.average.nmat=0.05, min.max.cluster.average.smat=0.01) {
   if (!requireNamespace("velocyto.R")) {
     stop("You need to install 'velocyto.R' package to be able to use this function")
   }
 
+  if(!is.null(clustering)) {
+    if(clustering=="leiden") {
+      groups <- con$clusters$leiden$groups
+    } else if(clustering=="walktrap") {
+      groups <- con$clusters$walktrap$groups
+    } else {
+      stop("Unknown 'clustering'.")
+    }
+  }
+
+  if(is.null(groups)) {
+    stop("Need 'groups' argument.")
+  } else if(!any(names(groups) %in% names(V(con$graph)))) {
+    stop("'groups' not defined for graph object.")
+  }
+
+  cell.colors <- fac2col(groups)
+  emb <- con$embedding
```

Comment:
Would be nice to check on null

Reply:
I added that in the commit 89701492b7c697cb8b90f2018a326fe187c2d170